sdk/Sdk.Generators/FunctionMethodVisibility.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.Functions.Worker.Sdk.Generators
+{
+    /// <summary>
+    /// Represents the visibility of an "azure function" method and its parent classes.
+    /// </summary>
+    internal enum FunctionMethodVisibility
+    {
+        /// <summary>
+        /// The method and it's parent classes are public.
+        /// </summary>
+        PublicAndVisible,
+
+        /// <summary>
+        /// The method is public, but one or more of its parent classes are not public.
+        /// </summary>
+        PublicButContainingTypeNotVisible,
```

Comment:
Curious why this is treated differently / needs to be its own state

Reply:
If one writes a public method, but the class it resides is internal, we want to treat them differently when executing. We will not be able to use the src gen version executor approach in that case as we will not be able to access the type from the entry assembly. In that case, we will fallback to the existing reflection based executor. 
The legacy version of executor supports this use case. So our new executor  (src gen version) should also work for that case.